datacube/metadata/_eo3converter.py

```diff
@@ -164,6 +163,38 @@ def _to_grid(gbox: GeoBox) -> dict[str, Any]:
     return {"shape": gbox.shape.yx, "transform": gbox.transform[:6]}  # type: ignore[index]
 
 
+def _to_eo3_properties(properties: dict[str, Any]) -> dict[str, Any]:
+    """
+    Convert EO3 properties dictionary to the Stac equivalent.
+    """
+
+    def _to_eo3_type(key: str, value):
+        if key == "instruments":
+            if len(value) > 0:
+                return "_".join([i.upper() for i in value])
+            else:
+                return None
+        else:
+            return value
+
+    prop = {
+        STAC_TO_EO3_RENAMES.get(key, key): _to_eo3_type(key, val)
+        for key, val in properties.items()
+    }
+    creation_time = (
+        properties.get("odc:processing_datetime")
+        or properties.get("created")
+        or properties.get("datetime")
+    )
+    if prop.get("odc:processing_datetime") is None and creation_time:
```

Comment:
If you just want a membership check of the key, I think `key in dict` or `key not in dict` works.